core/src/main/java/org/apache/druid/guice/GuiceInjectableValues.java

```diff
@@ -50,7 +59,21 @@ public Object findInjectableValue(
     // Currently we should only be dealing with `Key` instances, and anything more advanced should be handled with
     // great care
     if (valueId instanceof Key) {
-      return injector.getInstance((Key) valueId);
+      try {
+        return injector.getInstance((Key) valueId);
+      }
+      catch (ConfigurationException ce) {
+        // check if nullable annotation is present for this
+        if (nullables.get().contains((Key) valueId)) {
```

Comment:
The point of caching these is so that we only have Guice generate the exception once.  Putting this check inside of the catch means that we don't get any benefit.  This check needs to be done before we even ask the injector to create the instance.